lighthouse-core/audits/lazy-third-party.js

```diff
@@ -0,0 +1,189 @@
+    const entitySummaries = new Map();
+
+    for (const [url, urlSummary] of byURL) {
+      const entity = thirdPartyWeb.getEntity(url);
+      const product = thirdPartyWeb.getProduct(url);
+      if (!entity || mainEntity && entity.name === mainEntity.name) continue;
+
+      /** @type {Map<string, ProductSummary>} */
+      const productSummaries = entitySummaries.get(entity.name) || new Map();
+      if (product && product.facades && product.facades.length) {
+        // Record new url if product has a facade
+        const productSummary = productSummaries.get(product.name) || {
+          product,
+          cutoffTime: Infinity,
+          urlSummaries: new Map(),
+        };
+
+        productSummary.urlSummaries.set(url, urlSummary);
+        productSummary.cutoffTime = Math.min(productSummary.cutoffTime, urlSummary.endTime);
+
+        productSummaries.set(product.name, productSummary);
+      } else {
+        // If the url does not have a facade but one or more products on its entity do,
+        // we still want to record this url because it was probably fetched by a product with a facade.
+        for (const productSummary of productSummaries.values()) {
+          if (urlSummary.startTime < productSummary.cutoffTime) continue;
+          productSummary.urlSummaries.set(url, urlSummary);
+        }
+      }
+      entitySummaries.set(entity.name, productSummaries);
+    }
+
+    const allProductSummaries = [];
+    for (const productSummaries of entitySummaries.values()) {
+      for (const productSummary of productSummaries.values()) {
+        allProductSummaries.push(productSummary);
+      }
+    }
+    return allProductSummaries;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const settings = context.settings || {};
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const mainResource = await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
+    const mainEntity = thirdPartyWeb.getEntity(mainResource.url);
+    const tasks = await MainThreadTasks.request(trace, context);
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+      settings.throttling.cpuSlowdownMultiplier : 1;
+    const {byURL} = ThirdPartySummary.getSummaries(networkRecords, tasks, multiplier);
+    const productSummaries = LazyThirdParty.getProductSummaries(byURL, mainEntity);
+
+    const summary = {wastedBytes: 0, wastedMs: 0};
+
+    /** @type LH.Audit.Details.TableItem[] */
+    const results = [];
+    for (const productSummary of productSummaries) {
+      const product = productSummary.product;
+      if (!product.facades || !product.facades.length) continue;
+
+      // The first facade should always be the best one
```

Comment:
```suggestion
      // The first facade should always be the best one.
```

this is guaranteed by third-party-web?

Reply:
Yes